testing/proftpd/APKBUILD

```diff
@@ -0,0 +1,147 @@
+# Contributor: Valery Kartel <valery.kartel@gmail.com>
+# Contributor: Elizabeth Jennifer Myers <elizabeth@sporksirc.net>
+# Maintainer: Valery Kartel <valery.kartel@gmail.com>
+pkgname=proftpd
+pkgver=1.3.6_rc4
+pkgrel=0
+_pkgver=${pkgver/_rc/rc}
+pkgdesc="Highly configurable GPL-licensed FTP server software"
+url="http://www.proftpd.org/"
+pkgusers="proftpd"
+pkggroups="proftpd"
+arch="all"
+license="GPL"
+depends=""
+depends_dev="perl"
+makedepends="bash pcre-dev libressl-dev sqlite-dev mariadb-dev unixodbc-dev
+	postgresql-dev openldap-dev geoip-dev net-snmp-dev"
+install="$pkgname.pre-install"
+_modules="auth_file ban copy ctrls_admin deflate delay dnsbl dynmasq exec geoip facl facts ident
+	ifsession:=zz ifversion ldap load log_forensic qos quotatab:-- quotatab_file:quotatab
+	quotatab_ldap:quotatab quotatab_radius:quotatab quotatab_sql:quotatab,sql radius ratio
+	readme rewrite shaper site_misc sql:-- sql_passwd:sql sftp_sql:sftp,sql sql_sqlite:sql
+	sql_mysql:sql sql_odbc:sql sql_postgres:sql tls tls_shmcache:tls unique_id"
+for _module in $_modules; do
+	[ -z "${_module##*:*}" ] && eval _load_mod_${_module%:*}=${_module#*:}
+	subpackages="$subpackages $pkgname-mod_${_module%:*}:_module"
+	_shared_mods="${_shared_mods:+$_shared_mods:}mod_${_module%:*}"
+done
+subpackages="$pkgname-dev $pkgname-doc $pkgname-utils
+	$pkgname-mod_sftp $pkgname-mod_snmp $subpackages
+	"
+source="ftp://ftp.proftpd.org/distrib/source/$pkgname-$_pkgver.tar.gz
+	$pkgname.logrotate
+	$pkgname.initd
+	mod_delay.conf
+	mod_load.conf
+	mod_sftp.conf
+	mod_tls_shmcache.conf
+	libressl.patch
+	"
+builddir="$srcdir/$pkgname-$_pkgver"
+
+build() {
+	cd "$builddir"
+	./configure \
+		--prefix=/usr \
+		--sysconfdir=/etc/$pkgname \
+		--localstatedir=/run/$pkgname \
+		--libexecdir=/usr/lib/$pkgname \
+		--disable-strip \
+		--disable-pam \
+		--enable-dso \
+		--enable-facl \
+		--enable-pcre \
+		--enable-ipv6 \
+		--enable-ctrls \
+		--enable-openssl \
+		--with-shared=mod_sftp:mod_snmp:$_shared_mods \
+		|| return 1
+	make || return 1
+}
+
+package() {
+	cd "$builddir"
+	make DESTDIR="$pkgdir" install || return 1
+	sed -i	-e "5 a Include /etc/$pkgname/modules.d/*.conf" \
+		-e "5 a Include /etc/$pkgname/conf.d/*.conf\n" \
+		-e "s/nobody/$pkgusers/" \
+		-e "s/nogroup/$pkggroups/" \
+		"$pkgdir"/etc/$pkgname/$pkgname.conf || return 1
+	install -d "$pkgdir"/var/log/$pkgname
+	install -d "$pkgdir"/etc/$pkgname/conf.d
+	install -d "$pkgdir"/etc/$pkgname/modules.d
+	install -dm750 -o ftp -g ftp "$pkgdir"/var/lib/ftp
+	install -Dm755 "$srcdir"/$pkgname.initd "$pkgdir"/etc/init.d/$pkgname
+	install -Dm644 "$srcdir"/$pkgname.logrotate "$pkgdir"/etc/logrotate.d/$pkgname
+	rm -fr "$pkgdir"/run "$pkgdir"/usr/share/locale \
+		"$pkgdir"/usr/sbin/in.$pkgname
+}
+
+dev() {
+	default_dev
```

Comment:
Unfortunately APKBUILD does not run with `set -e` (yet, it’s on our TODO list to change it), so every statement that may fail, but should not fail, and is not the last statement of the function, should end with `|| return 1`.